### PR TITLE
Github workflows develop

### PR DESCRIPTION
**Pull request by DavidMansolino**
This pull-request fixes issue #2126

  - [x] Create a worfklow for linux nightlies on develop
  - [x] Create a worfklow for macOS nightlies on develop
  - [ ] Create a worfklow for Windows nightlies on develop
  - [x] Create a test to make sure default and develop stay synchronized.

### CODE REVIEW

**Comment by DavidMansolino** [comment]
Difference between main and develop workflows: 2bea681003f48527cae606a2ab4b78db896bad3e

File: .github/workflows/test_suite_linux_develop.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+jobs:
+  cleanup-runs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: rokroskar/workflow-run-cleanup-action@master
+      env:
+        GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+  build:
+    needs: cleanup-runs
+    if: ${{ github.event_name == 'push' || github.event_name == 'schedule' || contains(github.event.pull_request.labels.*.name, 'test distribution') || contains(github.event.pull_request.labels.*.name, 'test suite') || contains(github.event.pull_request.labels.*.name, 'test ros') || contains(github.event.pull_request.labels.*.name, 'test world update') }}
```
Comment (DavidMansolino) [inline .github/workflows/test_suite_linux_develop.yml:19]: Some parts such as this `if` could be removed/simplified on the develop workflow, but I prefer to keep them to keep the difference between workflows as small as possible.

File: .github/workflows/test_workflows_synchronization.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    - '.github/workflows/**'
+
+jobs:
+  test-doc:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Tests
+      run: |
+        if [[ "$(diff .github/workflows/test_suite_linux.yml .github/workflows/test_suite_linux_develop.yml | grep -c -e \"^<\" -e \"^>\")" -ne "13" ]]; then echo Linux Workflows not synchronized; exit -1; fi
```
Comment (DavidMansolino) [inline .github/workflows/test_workflows_synchronization.yml:15]: Checking the number of diff lines is quite simple, but should be sufficient to detect when one workflow is updated and not the other.

Comment (DavidMansolino) [reply .github/workflows/test_workflows_synchronization.yml]: Not sure if it is better to create a dedicated workflow or merge it with the 'source' one?

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**